src/iop/sigmoid.c

```diff
@@ -355,27 +411,81 @@ static void _calculate_adjusted_primaries(const dt_iop_sigmoid_data_t *const mod
+                       base_profile->matrix_out_transposed);
+    mat3SSEinv(base_to_pipe, pipe_to_base);
+  }
+  else
+  {
+    // Special case: if pipe and base profile are the same, pipe_to_base is an identity matrix.
+    for(size_t i = 0; i < 4; i++)
+    {
+      for(size_t j = 0; j < 4; j++)
+      {
+        if(i == j && i < 3)
+        {
+          pipe_to_base[i][j] = 1.f;
+          base_to_pipe[i][j] = 1.f;
+        }
+        else
+        {
+          pipe_to_base[i][j] = 0.f;
+          base_to_pipe[i][j] = 0.f;
+        }
+      }
+    }
+  }
+
+  // Rotated, scaled primaries are calculated based on the "base profile"
   float custom_primaries[3][2];
   for(size_t i = 0; i < 3; i++)
-    dt_rotate_and_scale_primary(pipe_work_profile, 1.f - module_data->inset[i], module_data->rotation[i], i,
+    dt_rotate_and_scale_primary(base_profile, 1.f - module_data->inset[i], module_data->rotation[i], i,
                                 custom_primaries[i]);
 
   dt_colormatrix_t custom_to_XYZ;
-  dt_make_transposed_matrices_from_primaries_and_whitepoint(custom_primaries, pipe_work_profile->whitepoint,
+  dt_make_transposed_matrices_from_primaries_and_whitepoint(custom_primaries, base_profile->whitepoint,
                                                             custom_to_XYZ);
-  dt_colormatrix_mul(pipe_to_rendering, custom_to_XYZ, pipe_work_profile->matrix_out_transposed);
+  dt_colormatrix_mul(base_to_rendering, custom_to_XYZ, base_profile->matrix_out_transposed);
 
   for(size_t i = 0; i < 3; i++)
   {
     const float scaling = 1.f - module_data->purity * module_data->inset[i];
-    dt_rotate_and_scale_primary(pipe_work_profile, scaling, module_data->rotation[i], i, custom_primaries[i]);
+    dt_rotate_and_scale_primary(base_profile, scaling, module_data->rotation[i], i, custom_primaries[i]);
   }
 
-  dt_make_transposed_matrices_from_primaries_and_whitepoint(custom_primaries, pipe_work_profile->whitepoint,
+  dt_make_transposed_matrices_from_primaries_and_whitepoint(custom_primaries, base_profile->whitepoint,
                                                             custom_to_XYZ);
   dt_colormatrix_t tmp;
-  dt_colormatrix_mul(tmp, custom_to_XYZ, pipe_work_profile->matrix_out_transposed);
-  mat3SSEinv(rendering_to_pipe, tmp);
+  dt_colormatrix_mul(tmp, custom_to_XYZ, base_profile->matrix_out_transposed);
+  dt_colormatrix_t rendering_to_base;
+  mat3SSEinv(rendering_to_base, tmp);
+  dt_colormatrix_mul(rendering_to_pipe, rendering_to_base, base_to_pipe);
+}
+
+static dt_colorspaces_color_profile_type_t _get_base_profile_type(dt_iop_sigmoid_base_primaries_t base_primaries)
+{
+  if(base_primaries == DT_SIGMOID_SRGB)
+    return DT_COLORSPACE_SRGB;
+
+  if(base_primaries == DT_SIGMOID_DISPLAY_P3)
+    return DT_COLORSPACE_DISPLAY_P3;
+
+  return DT_COLORSPACE_LIN_REC2020;
+}
+
+static const dt_iop_order_iccprofile_info_t * _get_base_profile(struct dt_develop_t *dev,
```

Comment:
`
static const _iop_order_iccprofile_info_t * ...`

Reply:
`dt_` prefix is the prefix of the return type.

Reply:
Sorry :)